clusterversion: start 25.3 versions

RaduBerinde opened the pull request:
**Checklist**:

- [x] Add version key constant for new release (e.g. `V24_2`), equal to `Latest`.
  Note that we do NOT update the `PreviousRelease` constant. We can only update the
  `PreviousRelease` when an RC is published, which is needed for upgrade
  roachtests.

- [x] Add final version for the previous release (e.g. `V24_1`).

- [x] Add start version (e.g. `V24_2_Start` with version `24.1-2`) and add a new
  first upgrade for it (in `upgrades/upgrades.go`).

- [x] Update `SystemDatabaseSchemaBootstrapVersion` in
  `pkg/sql/catalog/systemschema/system.go` to the start version just created.

- [x] Update `roachpb.successorSeries` map and update `TestReleaseSeriesSuccessor`

- [x] Update `pkg/build/version.txt` to the new version (e.g. `v24.2.0-alpha.00000000`)

- [x] Update the `scplan` rules in `pkg/sql/schemachanger/scplan/internal/rules`:
  - copy the contents of `current` into a new release directory for the previous version
    (e.g. `release_24_1`)
  - change package name in all files (including `BUILD.bazel`), and update the
    version in `current/helpers.go`
  - update `rulesForReleases` in `scplan/plan.go`
  - rewrite the test outputs: `./dev test pkg/sql/schemachanger/scplan/internal/rules/... --rewrite`
  - rewrite `TestDeclarativeRules` output: `./dev test pkg/cli -f DeclarativeRules --rewrite`

- [x] Update releases file:
  ```
  bazel build //pkg/cmd/release:release
  _bazel/bin/pkg/cmd/release/release_/release update-releases-file
  ```
  Note that the forked release version will disappear from the releases file. This
  is what we want; if it is here, we will attempt to run upgrade tests against it.

- [x] Regenerate expected test data as needed (usually
  `pkg/sql/catalog/systemschema_test` and some logictests).

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/144553)
<!-- Reviewable:end -->

RaduBerinde committed:
clusterversion: start 25.3 versions

**Checklist**:

- [x] Add version key constant for new release (e.g. `V24_2`), equal to `Latest`.
  Note that we do NOT update the `PreviousRelease` constant. We can only update the
  `PreviousRelease` when an RC is published, which is needed for upgrade
  roachtests.

- [x] Add final version for the previous release (e.g. `V24_1`).

- [x] Add start version (e.g. `V24_2_Start` with version `24.1-2`) and add a new
  first upgrade for it (in `upgrades/upgrades.go`).

- [x] Update `SystemDatabaseSchemaBootstrapVersion` in
  `pkg/sql/catalog/systemschema/system.go` to the start version just created.

- [x] Update `roachpb.successorSeries` map and update `TestReleaseSeriesSuccessor`

- [x] Update `pkg/build/version.txt` to the new version (e.g. `v24.2.0-alpha.00000000`)

- [x] Update the `scplan` rules in `pkg/sql/schemachanger/scplan/internal/rules`:
  - copy the contents of `current` into a new release directory for the previous version
    (e.g. `release_24_1`)
  - change package name in all files (including `BUILD.bazel`), and update the
    version in `current/helpers.go`
  - update `rulesForReleases` in `scplan/plan.go`
  - rewrite the test outputs: `./dev test pkg/sql/schemachanger/scplan/internal/rules/... --rewrite`
  - rewrite `TestDeclarativeRules` output: `./dev test pkg/cli -f DeclarativeRules --rewrite`

- [x] Update releases file:
  ```
  bazel build //pkg/cmd/release:release
  _bazel/bin/pkg/cmd/release/release_/release update-releases-file
  ```
  Note that the forked release version will disappear from the releases file. This
  is what we want; if it is here, we will attempt to run upgrade tests against it.

- [x] Regenerate expected test data as needed (usually
  `pkg/sql/catalog/systemschema_test` and some logictests).

Epic: none
Release note: None

rail reviewed (approved):
Thank you!

celiala commented:
Since this could potentially cause issue with master-to-master upgrades, we should check in with the DRP team, on whether they plan to perform any between now and Eng completion. fyi @RaduBerinde @rail

celiala reviewed (approved):
Thank you - LGTM!

RaduBerinde commented:
TFTRs!

bors r+
